relevanceai/apps/explorer_app/app.py

```diff
@@ -0,0 +1,87 @@
+from typing import Dict, List, Optional, Union
+from relevanceai.apps.explorer_app.base import ExplorerBase
+
+class ExplorerApp(ExplorerBase):
+    def preview_fields(self, fields=[], append=True):
+        self._append_or_replace("default-preview-centroids-columns", fields, default_value=[])
+
+    #Search and filter section
+    def facets(self, fields=[], append=True):
+        self._append_or_replace("filtering-facets", fields, default_value=[])
+
+    def sort(self, metrics=[], append=True):
+        if "key-metrics" not in self.config:
+            self.config["key-metrics"]["previewQuery"] = {"groupby" : [], "metrics" : []}
+        if append:
+            self.config["key-metrics"]["previewQuery"]["metrics"] += self.dataset._clean_metrics(metrics, name_prefix="")[0]
+        else:
+            self.config["key-metrics"]["previewQuery"]["metrics"] = self.dataset._clean_metrics(metrics, name_prefix="")[0]
+
+    def sort_default(self, metric_name):
+        if "key-metrics" not in self.config or not self.config["key-metrics"]:
+            raise TypeError("'sort' needs to be specified to use 'sort_default'")
+        if isinstance(metric_name, str):
+            self.config['sort-default'] = self.dataset._return_sort_in_metrics(
+                metric_name, 
+                self.config["key-metrics"]["previewQuery"]["metrics"]
+            )
+        else:
+            raise TypeError("'sort_default' needs to be of type string.")
+
+    def search_fields(self, fields=[]):
+        self.config["text-fields-to-search"] = fields
+    
+    def vector_search_fields(self, vector_fields=[]):
+        self.config["vector-fields-to-search"] = vector_fields
+    
+    def search_min_relevance(self, min_relevance):
+        self.config["search-minimum-relevance"] = min_relevance
+    
+    #Chart section
+    def chart(
+        self, 
+        groupby:List[Dict]=[], 
```

Comment:
@boba-and-beer should this be `groupby: Optional[List[Dict]]=None`, and then further done hand if `groupby is None`